core/src/test/scala/weaponregex/parser/ParserJVMTest.scala

```diff
@@ -25,9 +25,8 @@ class ParserJVMTest extends ParserTest {
 
   test("Parse character class with nested character classes") {
     val pattern = "[[a-z][^A-Z0-9][01234]]"
-    val parsedTree = Parser(pattern, parserFlavor).get
+    val parsedTree = Parser(pattern, parserFlavor).get.asInstanceOf[CharacterClass]
```

Comment:
Will this casting result in casting exceptions instead of test failures when the parser produces incorrect node type? I think we should still keep the `assert` for the node type to get some clues of what was produced by the parser.

Reply:
Yeah good point, it'll only show the class instead of the whole parsed element. I've made a 'smarter' cast that produces something like this:

```
Failed:
- weaponregex.parser.ParserJVMTest:
  * weaponregex.parser.ParserJVMTest.Parse concat of characters - munit.FailException: /home/hugovr/ws/weapon-regex/core/src/test/scala/weaponregex/parser/ParserTest.scala:36 Expected Or but got Concat
35:    val pattern = "hello"
36:    val parsedTree = Parser(pattern, parserFlavor).get.to[Or]
37:
Clues {
  tree: RegexTree = Concat(
    nodes = List(
      Character(
...
```

Reply:
Perfect 👌